android/src/com/mapswithme/util/Utils.java

```diff
@@ -105,6 +106,16 @@ public static void keepScreenOn(boolean enable, Window w)
       w.clearFlags(android.view.WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
   }
 
+  public static void showOnLockScreen(boolean enable, Activity activity)
+  {
+    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1)
+      activity.setShowWhenLocked(enable);
+    else if (enable)
+      activity.getWindow().addFlags(WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED);
+    else
+      activity.getWindow().clearFlags(WindowManager.LayoutParams.FLAG_SHOW_WHEN_LOCKED);
```

Comment:
Flag is deprecated because `setShowWhenLocked` has been added.
I use it just above for API 27+. This approach allows to support all Android versions.

I use Android 11, I will share video in the evening.

Reply:
I added video in the description of pull request

Reply:
Thanks for the video. I still don't see any UX changes on my Android 11/Android 12 devices. I lock the screen while running the app and nothing happens. I see absolutely no difference with or without flag enabled.

Reply:
I was able to get it working after playing with "Lock" settings on Samsung. Good feature, thanks!